app/models/concerns/user_totp_methods.rb

```diff
@@ -37,6 +38,7 @@ def enable_totp!(seed, level)
     if mfa_device_count_one?
       self.mfa_level = level
       self.mfa_recovery_codes = Array.new(10).map { SecureRandom.hex(6) }
+      self.hashed_mfa_recovery_codes = mfa_recovery_codes.map { |code| BCrypt::Password.create(code) }
```

Comment:
On top of a migration to remove `mfa_recovery_codes` from the model, I assume the other few instances of where `mfa_recovery_codes` is used like this will be removed?

If so, perhaps changing the line to 
```ruby
self.hashed_mfa_recovery_codes = Array.new(10).map { BCrypt::Password.create(SecureRandom.hex(6)) }
```
may make things simpler to port over after.